src/Astreiko.Homework9.Nbrb.by/API Client/APIClient.cs

```diff
@@ -0,0 +1,131 @@
+
+namespace Astreiko.Homework9.Nbrb.by.API_Client
+{
+    public class APIClient
+    {
+        /// <summary>
+        /// Dictionary for currencies (save code currency and internal code)
+        /// </summary>
+        private Dictionary<int, int> dictionaryCurrencies;
+
+        private static HttpClient httpClient;
+
+        public APIClient()
+        {
+            dictionaryCurrencies = new Dictionary<int, int>();
+            httpClient = new HttpClient();
+        }
+
+        /// <summary>
+        /// Getting full json file with all currencies
+        /// </summary>
+        /// <returns>list Currencies</returns>
+        private async Task<List<Currency>> GetAllCurrenciesAsync()
+        {
+            //HttpClient httpClient = new HttpClient();
+            //string request = "https://www.nbrb.by/api/exrates/currencies";
+            //HttpResponseMessage response = (await httpClient.GetAsync(request)).EnsureSuccessStatusCode();
+            //string responseBody = await response.Content.ReadAsStringAsync();
+
+            //return JsonConvert.DeserializeObject<List<Currencies>>(responseBody);
+            
+            return await httpClient.GetFromJsonAsync<List<Currency>>("https://www.nbrb.by/api/exrates/currencies");
+        }
+
+        /// <summary>
+        /// Receiving currency as a short list
+        /// </summary>
+        /// <returns>list short currencies</returns>
+        /// 
+        public async Task<List<ShortCurrency>> GetShortCurrenciesAsync()
+        {
+            var listCurrencies = (await GetAllCurrenciesAsync()).ToList();
+
+            CreateDictionaryCurrencies(listCurrencies);
+
+            return listCurrencies.Where(x => x.Cur_DateEnd > DateTime.Now)
+                .OrderBy(y => y.Cur_Code)
+                .Select(c => new ShortCurrency()
+                {
+                    Name = c.Cur_Name,
+                    Abbreviation = c.Cur_Abbreviation,
+                    Code = c.Cur_Code
+                })
+                .ToList();                       
+        }
+
+        /// <summary>
+        /// Create dictionary for currencies
+        /// </summary>
+        /// <param name="listCurrencies">list currencies</param>
+        private void CreateDictionaryCurrencies(List<Currency> listCurrencies)
+        {
+            dictionaryCurrencies.Clear();
+
+            foreach (var currency in listCurrencies.Where(x => x.Cur_DateEnd > DateTime.Now))
+            {
+                dictionaryCurrencies.Add(currency.Cur_Code, currency.Cur_ID);
+            }
+        }
+
+        /// <summary>
+        /// Task for get Rates
+        /// </summary>
+        /// <param name="forDate">Date currency</param>
+        /// <param name="codeCurrency">Code currency</param>
+        /// <returns>Task</returns>
+        private async Task<Rate> GetRateOnDateAsync(DateTime forDate, int codeCurrency)
+        {
+            CreateDictionaryCurrencies((await GetAllCurrenciesAsync()).ToList());
```

Comment:
I think you should not recreate dictionary each time, it is enough to get it once (it does not contradict to the purpose of api client because requesting rates by internal code is a kind of api specifics)